### PR TITLE
Exception parameters are intrinsically NonNull

### PR DESCRIPTION
**Pull request by wmdietl**
Relates to the discussion in #347.
This PR updates the spec and completes the conformance tests.

### CODE REVIEW

File: conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/notnullmarked/Other.java [hunk 1]
```diff
@@ -65,6 +65,12 @@ void catchParameters() {
       // test:irrelevant-annotation:NonNull
```
Comment (wmdietl) [inline conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/notnullmarked/Other.java:65]: Should there be different tests for "irrelevant" vs. "intrinsically non-null"?

Comment (netdpb) [reply conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/notnullmarked/Other.java:65]: We've been using `irrelevant-annotation` for all cases where the annotation is unrecognized or inapplicable, without requiring tools to distinguish.

File: conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/notnullmarked/Other.java [hunk 2]
```diff
@@ -65,6 +65,12 @@ void catchParameters() {
...
     } catch (@NonNull RuntimeException e) {
     }
+    try {
+    } catch (RuntimeException e) {
+      // test:name:Intrinsically NonNull exception parameter cannot be assigned null
+      // test:cannot-convert:null? to RuntimeException
```
Comment (netdpb) [inline conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/notnullmarked/Other.java:71]:
```suggestion
      // test:cannot-convert:null? to RuntimeException!
```

File: conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/nullmarked/Other.java [hunk 3]
```diff
@@ -65,6 +65,12 @@ void catchParameters() {
       // test:irrelevant-annotation:NonNull
     } catch (@NonNull RuntimeException e) {
     }
+    try {
+    } catch (RuntimeException e) {
+      // test:name:Intrinsically NonNull exception parameter cannot be assigned null
+      // test:cannot-convert:null? to RuntimeException
```
Comment (netdpb) [inline conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/nullmarked/Other.java:71]:
```suggestion
      // test:cannot-convert:null? to RuntimeException!
```

File: conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/nullunmarked/Other.java [hunk 4]
```diff
@@ -67,6 +67,12 @@ void catchParameters() {
       // test:irrelevant-annotation:NonNull
     } catch (@NonNull RuntimeException e) {
     }
+    try {
+    } catch (RuntimeException e) {
+      // test:name:Intrinsically NonNull exception parameter cannot be assigned null
+      // test:cannot-convert:null? to RuntimeException
```
Comment (netdpb) [inline conformance-tests/src/assertions/java/org/jspecify/conformance/tests/irrelevantannotations/nullunmarked/Other.java:73]:
```suggestion
      // test:cannot-convert:null? to RuntimeException!
```

**Review by netdpb — CHANGES_REQUESTED** [review]

**Review by netdpb — APPROVED** [review]

**Comment by wmdietl** [comment]
> @wmdietl are you waiting for something before you squash-and-merge?

Thanks for the ping.

**Pull request merged**